### PR TITLE
Modified tag feature (emergency and close friend)

### PR DESCRIPTION
**Pull request by tanteckfang**
I have edited the tag feature such that it can only accept 2 emergency tag, and it will automatically convert it to 'EMERGENCY'. I also improve the feature such that if someone type 't/ cf' it will add a 'Close Friend' tag to the contact

### CODE REVIEW

**Review by Carlintyj — COMMENTED** [review]
Hello! Next time please continue to branch off the master instead of a branch of another PR. 

I like how you made a "cf" shortcut and capitalise "EMERGENCY" when adding of the tag. Also you limited the number of emergency contacts good job!

**Review by leontan2 — CHANGES_REQUESTED** [review]
Hi Teck Fang, nice progress with the code!
 
However, upon review, I've noticed that the code does not fully meet the specified requirements in terms of tag command.
It seems that tags allowed by the user are not addressed as expected.

Would it be better for the user to only be able to tag their Friends, Close Friends and Emergency, perhaps using an ENUM class would be more appropriate since there are only 3 categories for the people that are added to the addressbook.

 Also, remember to assign yourself as the assignee and assign the development to the issue for the PR tracker!

**Comment by tanteckfang** [comment]
Hi Leon and Carlin, thanks for your suggestion, tryuly appreciate it. I have made some changes such that it can only allow three types of tags which are Friends, Close Friends and EMERGENCY

File: src/main/java/seedu/address/logic/commands/AddCommand.java
Comment (Carlintyj) [inline src/main/java/seedu/address/logic/commands/AddCommand.java:1]: Hi TF, are you able to change line 39 and 40?

File: src/main/java/seedu/address/model/tag/Tag.java [hunk 1]
```diff
@@ -9,8 +9,8 @@
  */
 public class Tag {
 
-    public static final String MESSAGE_CONSTRAINTS = "Tags names should be alphanumeric";
-    public static final String VALIDATION_REGEX = "\\p{Alnum}+";
+    public static final String MESSAGE_CONSTRAINTS =
+            "Only three kind of tag names are allowed: Friends, Close Friends (or cf), and EMERGENCY'";
```
Comment (Carlintyj) [inline src/main/java/seedu/address/model/tag/Tag.java:13]: Is it "Friends" and "Close Friends" or "Friend" and "Close Friend"

**Comment by joel-foo** [comment]
Hi, thanks for helping to make the changes. however, something minor, I'm not sure we should use "EMERGENCY" instead of "Emergency". I think standardising the tags to be the same case format looks better on the UI. What do you guys think?

**Comment by joel-foo** [comment]
LGTM! If you have time in the future, you can trying creating an Enum class, as Leon mentioned. Will merge this now.

**Review by joel-foo — APPROVED** [review]
LGTM!

**Comment by joel-foo** [comment]
> Hi Teck Fang, nice progress with the code!
> 
> However, upon review, I've noticed that the code does not fully meet the specified requirements in terms of tag command. It seems that tags allowed by the user are not addressed as expected.
> 
> Would it be better for the user to only be able to tag their Friends, Close Friends and Emergency, perhaps using an ENUM class would be more appropriate since there are only 3 categories for the people that are added to the addressbook.
> 
> Also, remember to assign yourself as the assignee and assign the development to the issue for the PR tracker!

@leontan2, could you help remove your "request changes" so I can merge this? Thanks.

**Review by leontan2 — APPROVED** [review]
I have seen the changes made to the PR, thanks for making those changes
LGTM!

**Pull request merged**